openedx/core/djangoapps/profile_images/views.py

```diff
@@ -105,9 +108,24 @@ class ProfileImageView(APIView):
           the user exists or not.
     """
 
-    parser_classes = (MultiPartParser, FormParser)
+    parser_classes = (MultiPartParser, FormParser, TypedFileUploadParser)
     authentication_classes = (OAuth2AuthenticationAllowInactiveUser, SessionAuthenticationAllowInactiveUser)
     permission_classes = (permissions.IsAuthenticated, IsUserInUrl)
+    _supported_media_types = None
+
+    @property
+    def supported_media_types(self):
+        """
+        Dictionary that lists the supported mimetypes and maps them to allowed
+        file extensions.  Required by `TypedFileUploadParser`.
+        """
+        if not self._supported_media_types:
+            self._supported_media_types = {}
+            for type_ in IMAGE_TYPES.values():
+                self._supported_media_types.update(
+                    {mimetype: type_['extension'] for mimetype in type_['mimetypes']}
+                )
```

Comment:
I find nested comprehensions hard to read, so I usually unwrap the outer level.  Is your example a dict comprehension inside a set comprehension?